infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -129,10 +131,37 @@ public LogUnitServer(ServerContext serverContext) {
     /**
      * Service an incoming request for maximum global address the log unit server has written.
      */
-    @ServerHandler(type = CorfuMsgType.TAIL_REQUEST)
-    public void handleTailRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+    @ServerHandler(type = CorfuMsgType.LOG_TAIL_REQUEST)
+    public void handleLogTailRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        Long logTail = batchWriter.queryLogTail(msg.getEpoch());
+        r.sendResponse(ctx, msg, CorfuMsgType.LOG_TAIL_RESPONSE.payloadMsg(logTail));
+    }
+
+    /**
+     * Service an incoming request for all stream tails (also returns log global tail).
+     */
+    @ServerHandler(type = CorfuMsgType.TAILS_REQUEST)
+    public void handleTailsRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
         TailsResponse tails = batchWriter.queryTails(msg.getEpoch());
-        r.sendResponse(ctx, msg, CorfuMsgType.TAIL_RESPONSE.payloadMsg(tails));
+        r.sendResponse(ctx, msg, CorfuMsgType.TAILS_RESPONSE.payloadMsg(tails));
```

Comment:
we don't catch any exceptions here, is it ok?

Reply:
exceptions are caught by the NettyServerRouter